app/models/spree/payment.rb

```diff
@@ -154,8 +154,10 @@ def revoke_adjustment_eligibility
       return unless adjustment.try(:reload)
       return if adjustment.finalized?
 
-      adjustment.update_attribute(:eligible, false)
-      adjustment.finalize!
```

Comment:
I guess it doesn't do anything else other than changing the `state` column. Does it?

Reply:
In theory extra logic could be hooked in to the state_machine transitions in `Spree::Payment`, but we don't have any.